.gitlab-ci.yml

```diff
@@ -0,0 +1,118 @@
+before_script:
+  # set stack isolation
+  - export ISOLATION=buildpipeline${CI_PIPELINE_ID}${CI_BUILD_NAME}
+  - export COMPOSE_PROJECT_NAME=${ISOLATION}
+  - export TUPLE_C=$(expr ${CI_BUILD_ID} % 99)
+  - echo ${TUPLE_C}
+  # run docker-compose commands from tests environment
+  - cd tests
+  - cp .env-dist .env
+  - docker-compose config
+
+after_script:
+  - export ISOLATION=buildpipeline${CI_PIPELINE_ID}${CI_BUILD_NAME}
+  - export COMPOSE_PROJECT_NAME=${ISOLATION}
+  # run docker-compose commands from tests environment
+  - cd tests
+  - cp .env-dist .env
+  - docker-compose down -v --remove-orphans
+  - docker ps -f name=${ISOLATION}
+
+stages:
+  - travis
+  - test
+  - cleanup
+
+test:
+  stage: test
+  script:
+    - docker-compose up --build -d
+    - docker-compose run --rm php vendor/bin/phpunit -v --exclude caching,db,data --log-junit tests/_junit/test.xml
```

Comment:
The line I've commented has -rm argument, that according to the docs should remove container after the command execution, but ONLY when the container is NOT running in the background.

Line above (docker-compose up --build -d) stars container in background.
Does is make any sense to pass --rm flag for the RUN command in this case?